sql-non-prepared-plan-cache.md

```diff
@@ -66,24 +68,35 @@ The following example shows how to use the non-prepared plan cache:
 
+### Usage restrictions
+
 Due to the preceding risks and the fact that the execution plan cache only provides significant benefits for simple queries (if a query is complex and takes a long time to execute, using the execution plan cache might not be very helpful), TiDB has strict restrictions on the scope of non-prepared plan cache. The restrictions are as follows:
 
 - Queries or plans that are not supported by the [Prepared plan cache](/sql-prepared-plan-cache.md) are also not supported by the non-prepared plan cache.
-- Currently, only point get or range queries on a single table that contain `Scan`, `Selection`, or `Projection` operators are supported, such as `SELECT * FROM t WHERE a < 10 AND b in (1, 2)`.
-- Queries that contain complex operators such as `Agg`, `Limit`, `Window`, or `Sort` are not supported.
-- Queries that contain non-range query conditions are not supported, such as:
-    - `LIKE` is not supported, such as `c LIKE 'c%'`.
-    - `+` operation is not supported, such as `a+1 < 2`.
+- Queries that contain complex operators such as `Window` or `Sort` are not supported.
+- Queries that contain three or more `Join` tables or subqueries are not supported.
+- Queries that contain numbers or expressions directly after `ORDER BY` or `GROUP BY` are not supported, such as `ORDER BY 1` and `GROUP BY a+1`. Only `ORDER BY column_name` and `GROUP BY column_name` are supported.
 - Queries that filter on columns of `JSON`, `ENUM`, `SET`, or `BIT` type are not supported, such as `SELECT * FROM t WHERE json_col = '{}'`.
 - Queries that filter on `NULL` values are not supported, such as `SELECT * FROM t WHERE a is NULL`.
-- Queries with more than 50 parameters after parameterization are not supported, such as `SELECT * FROM t WHERE a in (1, 2, 3, ... 51)`.
+- Queries with more than 200 parameters after parameterization are not supported, such as `SELECT * FROM t WHERE a in (1, 2, 3, ... 201)`.
 - Queries that access partitioned tables, virtual columns, temporary tables, views, or memory tables are not supported, such as `SELECT * FROM INFORMATION_SCHEMA.COLUMNS`, where `COLUMNS` is a TiDB memory table.
-- Queries with hints, subqueries, or locks are not supported.
-- DML statements are not supported.
+- Queries with hints or bindings are not supported.
+
+After you enable this feature, the optimizer quickly assesses the query. If it does not meet the support conditions for non-prepared plan cache, the query goes through the regular optimization process.
+
+## Performance benefits
+
+In internal tests, enabling the non-prepared plan cache feature can achieve significant performance benefits in most TP scenarios. However, it also introduces some additional performance overhead, including determining whether the query is supported and parameterizing the query. If this feature cannot support the majority of queries in your workload, enabling it might actually impact performance.
```

Comment:
```suggestion
In internal tests, enabling the non-prepared plan cache feature can achieve significant performance benefits in most TP scenarios. However, it also introduces some additional performance overhead, including determining whether the query is supported and parameterizing the query. If this feature cannot support the majority of queries in your workload, enabling it might actually adversely affect performance.
```